### DISCUSSION TITLE
Extra server url config param cause an error

### DISCUSSION THREAD

**Discussion by boskiv** [strapi/strapi#6700] [Legacy Topics]
**Describe the bug**
Admin page does not opened when I use extra url param in server config

**Steps to reproduce the behavior**

1. Add usr param
```
module.exports = ({ env }) => ({
  host: env('HOST', '0.0.0.0'),
  port: env.int('PORT', 1337),
  url: 'api'
});
```

2. Run yarn build && yarn develop
3. Open localhost:1337/api/admin page
4. See error in browser log
```
GET http://localhost:1337/api/users-permissions/init 404 (Not Found)
```

**Expected behavior**
Admin page opens, routing works respecting /api param

**Code snippets**
Logs
```
[2020-06-17T02:05:40.127Z] debug GET /api/admin (8 ms) 200
[2020-06-17T02:05:40.155Z] debug GET runtime~main.30e246c4.js (7 ms) 200
[2020-06-17T02:05:40.156Z] debug GET main.2da58b00.chunk.js (2 ms) 200
[2020-06-17T02:05:40.587Z] debug GET /api/users-permissions/init (2 ms) 404
[2020-06-17T02:05:40.612Z] debug GET /favicon.ico (1 ms) 200
[2020-06-17T02:05:57.849Z] debug GET /api/admin (1 ms) 200
[2020-06-17T02:05:58.438Z] debug GET /api/users-permissions/init (2 ms) 404
[2020-06-17T02:05:58.478Z] debug GET /favicon.ico (0 ms) 200
```

**System**

- Node.js version: v12.18.0
- NPM version: 6.14.4
- Strapi version: 3.0.2
- Database: SQLite
- Operating system: MacOS 10.15.5

**Comment by boskiv** [comment]
I've also tried
```
admin: {
    build: {
      backend: '/api'
    }
  }
```
from https://strapi.io/documentation/v3.x/admin-panel/deploy.html#deploy-the-administration-panel-and-the-plugins-on-another-server-than-the-api

But no success

**Comment by derrickmehaffy** [comment]
> I've also tried
> 
> ```
> admin: {
>     build: {
>       backend: '/api'
>     }
>   }
> ```
> 
> from https://strapi.io/documentation/v3.x/admin-panel/deploy.html#deploy-the-administration-panel-and-the-plugins-on-another-server-than-the-api
> 
> But no success

Those docs are wrong and there is already a PR that addresses it: https://github.com/strapi/strapi/pull/6636

That `url` param in the server.js does nothing but tell Strapi the actual public URL, it will not prefix endpoints internally. If you set it to `https://example.com/api` Strapi will still respond on http://localhost:1337/whatever but it expects an upstream proxy like Nginx to send the request as https://example.com/api/whatever.

See the following docs for examples: https://strapi.io/documentation/v3.x/getting-started/deployment.html#optional-software-guides

**Comment by derrickmehaffy** [comment]
Will move this to a discussion as it's not a bug

**Reply by boskiv** [reply]
So there is now way to configure it in strapi, only at reverse proxy level ?
